pkg/controller/iam/managed.go

```diff
@@ -0,0 +1,261 @@
+			managed.WithExternalConnecter(&connecter{client: mgr.GetClient(), newSAS: newServiceAccountsAPI}),
+			managed.WithLogger(l.WithValues("controller", name)),
+			managed.WithInitializers(managed.NewNameAsExternalName(mgr.GetClient())),
+			managed.WithRecorder(event.NewAPIRecorder(mgr.GetEventRecorderFor(name)))))
+}
+
+// newServiceAccountsAPI returns a new IAM Admin Client (responsible for Service Account management).
+// Credentials must be passed as JSON encoded data.
+func newServiceAccountsAPI(ctx context.Context, credentials []byte) (*iamv1.ProjectsServiceAccountsService, error) {
+	service, err := iamv1.NewService(ctx, option.WithCredentialsJSON(credentials))
+	if err != nil {
+		return nil, err
+	}
+	return iamv1.NewProjectsService(service).ServiceAccounts, nil
+}
+
+type connecter struct {
+	client client.Client
+	newSAS func(ctx context.Context, creds []byte) (*iamv1.ProjectsServiceAccountsService, error)
+}
+
+// Connect sets up iam client using credentials from the provider
+func (c *connecter) Connect(ctx context.Context, mg resource.Managed) (managed.ExternalClient, error) {
+	cr, ok := mg.(*v1beta1.ServiceAccount)
+	if !ok {
+		return nil, errors.New(errNotServiceAccount)
+	}
+
+	p := &gcpv1alpha3.Provider{}
+	if err := c.client.Get(ctx, meta.NamespacedNameOf(cr.Spec.ProviderReference), p); err != nil {
+		return nil, errors.Wrap(err, errGetProvider)
+	}
+
+	if p.GetCredentialsSecretReference() == nil {
+		return nil, errors.New(errProviderSecretRef)
+	}
+
+	s := &corev1.Secret{}
+	n := types.NamespacedName{Namespace: p.Spec.CredentialsSecretRef.Namespace, Name: p.Spec.CredentialsSecretRef.Name}
+	if err := c.client.Get(ctx, n, s); err != nil {
+		return nil, errors.Wrap(err, errGetProviderSecret)
+	}
+
+	saAPI, err := c.newSAS(ctx, s.Data[p.Spec.CredentialsSecretRef.Key])
+	rrn := NewRelativeResourceNamer(p.Spec.ProjectID)
+	return &external{serviceAccounts: saAPI, rrn: rrn}, errors.Wrap(err, errNewClient)
+}
+
+type external struct {
+	serviceAccounts *iamv1.ProjectsServiceAccountsService
+	rrn             RelativeResourceNamer
+}
+
+func (e *external) Observe(ctx context.Context, mg resource.Managed) (managed.ExternalObservation, error) { // nolint:gocyclo
+	cr, ok := mg.(*v1beta1.ServiceAccount)
+	if !ok {
+		return managed.ExternalObservation{}, errors.New(errNotServiceAccount)
+	}
+
+	getRequest := e.serviceAccounts.Get(e.rrn.ResourceName(cr))
+	fromProvider, err := getRequest.Context(ctx).Do()
+	if err != nil {
+		if gcp.IsErrorNotFound(err) {
+			return managed.ExternalObservation{ResourceExists: false}, nil
+		}
+		return managed.ExternalObservation{}, errors.Wrap(err, errGet)
+	}
+
+	populateCRFromProvider(cr, fromProvider)
+	return managed.ExternalObservation{
+		ResourceExists:    true,
+		ResourceUpToDate:  isUpToDate(&cr.Spec.ForProvider, fromProvider),
+		ConnectionDetails: managed.ConnectionDetails{},
+	}, nil
+}
+
+// https://cloud.google.com/iam/docs/reference/rest/v1/projects.serviceAccounts/create
```

Comment:
I like this convention of including a convenient link to the underlying API you're calling. 👍